test/hotspot/jtreg/runtime/Nestmates/protectionDomain/TestDifferentProtectionDomains.java

```diff
@@ -110,10 +110,17 @@ public static void main(String[] args) throws Throwable {
 
         // Force the constructor to compile, which then triggers the nestmate
         // access check in the compiler thread, which leads to the original bug.
-        wb.enqueueMethodForCompilation(cons,  CompilerWhiteBoxTest.COMP_LEVEL_FULL_OPTIMIZATION);
-        while (!wb.isMethodCompiled(cons)) {
+        int optimizationLevel = CompilerWhiteBoxTest.COMP_LEVEL_FULL_OPTIMIZATION;
+        if (!wb.isC2OrJVMCIIncluded()) {
+            optimizationLevel = CompilerWhiteBoxTest.COMP_LEVEL_SIMPLE;
+        }
```

Comment:
I think it is better to check whether COMP_LEVEL_FULL_OPTIMIZATION is supported:
```
if (!wb.isMethodCompilable(optimizationLevel)) {
  optimizationLevel = CompilerWhiteBoxTest.COMP_LEVEL_ANY;
}
```